bitnami/jaeger/templates/agent/deployment.yaml

```diff
@@ -0,0 +1,205 @@
+            }
+
+            info "Connecting to the Cassandra instance $CQLSH_HOST:$CQLSH_PORT"
+            if ! retry_while "check_cassandra_keyspace_schema" 12 30; then
+              error "Could not connect to the database server"
+              exit 1
+            else
+              info "Connection check success"
+            fi
+        env:
+        - name: CQLSH
+          value: /opt/bitnami/cassandra/bin/cqlsh
+        - name: CQLSH_HOST
+          value: {{ printf "%s-cassandra" (include "common.names.fullname" .) }}
+        - name: CQLSH_PORT
+          value: {{ .Values.cassandra.containerPorts.cql | quote }}
+        - name: CASSANDRA_USERNAME
+          value: {{ .Values.cassandra.dbUser.user }}
+        - name: CASSANDRA_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ printf "%s-cassandra" (include "common.names.fullname" .) }}
+              key: cassandra-password
+        - name: CASSANDRA_KEYSPACE
+          value: {{ .Values.cassandra.keyspace }}
+      {{- if .Values.agent.initContainers }}
+      {{- include "common.tplvalues.render" (dict "value" .Values.agent.initContainers "context" $) | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: jaeger-agent
+          image: {{ include "jaeger.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          {{- if .Values.agent.containerSecurityContext.enabled }}
+          securityContext: {{- omit .Values.agent.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.command "context" $) | nindent 12 }}
+          {{- else if .Values.agent.command }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.agent.command "context" $) | nindent 12 }}
+          {{- else }}
+          command: ["/opt/bitnami/jaeger/bin/jaeger-agent"]
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
+          {{- else if .Values.agent.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.agent.args "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.agent.lifecycleHooks }}
+          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.agent.lifecycleHooks "context" $) | nindent 12 }}
+          {{- end }}
+          ports:
+            - name: binary
+              containerPort: {{ .Values.agent.service.ports.binary }}
+            - name: zipkin
+              containerPort: {{ .Values.agent.service.ports.zipkin }}
+            - name: admin
+              containerPort: {{ .Values.agent.service.ports.admin }}
```

Comment:
Also, you are using here `service.ports`, but as we are talking about the container, I would expect here `containerPorts` instead. The port configured in the service object does not have to be the same as the one configured in the container. A clear example is WordPress, where you can have the service port 80 but the container port 8080 (necessary as it is a non-root container)

Reply:
Refactored! [30f85ea](https://github.com/bitnami/charts/pull/13480/commits/30f85ea9714c8a1febc2a4d60718c45ce742af63)

Same as last conversation. Leaving open, feel free to close.